lib/internal/watch_mode/files_watcher.js

```diff
@@ -124,7 +126,28 @@ class FilesWatcher extends EventEmitter {
       this.#ownerDependencies.set(owner, dependencies);
     }
   }
+
+
+  #setupIPC(child) {
+    child._ipcMessages = {
```

Comment:
I think rather than adding properties to the child process we should use a week map:

```
	#ipcHandlers = new SafeWeakMap();
	#setupIPC(child) {
		const handlers = {
			__proto__: null,
			send: (message) => child.send(message),
			receive: (message) => process.send(message)
		};
		this.#ipcHandlers.set(child, handlers);
        process.on('message', handlers.send);
        child.on('message', handlers.receive);
	}
	destroyIPC(child) {
	  const handlers = this.#ipcHandlers.get(child)
  	  if (this.#wantsPassthroughIPC && handlers !== null) {
	        process.off('message', handlers.send);
	        child.off('message', handlers.receive);
	    }
    }
```

Reply:
I wasn't sure if using `WeakMap` in the internal code was appropriate - I'd never even heard of `SafeWeakMap` - but for sure that seems like a better idea.

Reply:
How strongly do you feel about the `send/receive` vs `parentToChild`? I find the send/receive naming confusing given that in both cases it is receiving, then sending a message :)

Reply:
not very strong, feel free to choose any name